components/sync15/src/state.rs

```diff
@@ -869,6 +869,9 @@ mod tests {
         let sm_seq_used_previous = vec!["WithPreviousState", "Ready"];
 
         // do the actual test.
+        // (clippy gets upset with `expected_states: &Vec<&str>` - wants it
+        // declared as `&[&str]` - which `assert_eq!()` then fails on!)
+        #[allow(clippy::ptr_arg)]
```

Comment:
Not sure what the story is here!

Reply:
Try Clippy's suggestion with `assert_eq!(state_machine.sequence.as_slice(), expected_states)`?

Reply:
heh - so that got closer, and then complained about `&[...]` vs `[...]` - basically telling me to remove the leading `*` (which is necessary as written) - which worked. Then, it works without the `.as_slice()` too!

I'm not sure how many clippy bugs are in the above, but all's well that ends well :)

Thanks!